feature-teacher/approve_outing/src/commonMain/kotlin/com.b1nd.dodam.approve-outing/ApproveOutScreen.kt

```diff
@@ -0,0 +1,335 @@
+                            )
+                        }
+                    }
+                },
+            )
+        }
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(DodamTheme.colors.backgroundNormal)
+                .padding(it),
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(horizontal = 16.dp),
+            ) {
+                Column {
+                    DodamSegmentedButton(
+                        segments = item,
+                        modifier = Modifier.padding(top = 12.dp),
+                    )
+                    DodamSegmentedButton(
+                        segments = gradeItem,
+                        modifier = Modifier.padding(top = 12.dp),
+                    )
+                    DodamSegmentedButton(
+                        segments = roomItem,
+                        modifier = Modifier.padding(top = 12.dp),
+                    )
+                }
+                DodamTextField(
+                    value = searchStudent,
+                    onValueChange = {
+                        searchStudent = it
+                    },
+                    label = "학생 검색",
+                    onClickRemoveRequest = {
+                        searchStudent = ""
+                    },
+                    modifier = Modifier,
+                )
+                when (val data = state.outPendingUiState) {
+                    OutPendingUiState.Error -> {}
+                    OutPendingUiState.Loading -> {}
+                    is OutPendingUiState.Success -> {
+                        val members =
+                            if (titleIndex == 0) data.outMembers else data.sleepoverMembers
+                        var filteredMemberList = if (gradeIndex == 0 && roomIndex == 0) {
+                            members
+                        } else if (gradeIndex == 0 && roomIndex != 0) {
+                            members.filter { studentData ->
+                                studentData.student.room == roomIndex
+                            }
+                        } else if (gradeIndex != 0 && roomIndex == 0) {
+                            members.filter { studentData ->
+                                studentData.student.grade == gradeIndex
+                            }
+                        } else {
+                            members.filter { studentData ->
+                                studentData.student.grade == gradeIndex && studentData.student.room == roomIndex
+                            }
+                        }
+                        if (searchStudent.isNotEmpty()) {
+                            filteredMemberList = filteredMemberList.filter {
+                                it.student.name.contains(searchStudent)
+                            }
+                        }
```

Comment:
```suggestion
                        val members = if (titleIndex == 0) data.outMembers else data.sleepoverMembers

                        val filteredMemberList = members.filter { studentData ->
                            when {
                                gradeIndex == 0 && roomIndex == 0 -> true
                                gradeIndex == 0 && roomIndex != 0 -> studentData.student.room == roomIndex
                                gradeIndex != 0 && roomIndex == 0 -> studentData.student.grade == gradeIndex
                                else -> studentData.student.grade == gradeIndex && studentData.student.room == roomIndex
                            }
                        }.let { filteredList ->
                            if (searchStudent.isNotEmpty()) {
                                filteredList.filter { it.student.name.contains(searchStudent) }
                            } else {
                                filteredList
                            }
                        }
```
다음과 같이 가독성을 개선해보시는건 어떨까요?

Reply:
좋은 충고 감사합니다.